### PR TITLE
Added pie stateDiagram gantt key words

### PR DESCRIPTION
**Pull request by nanjj**
And support publish all, add type to defcustom

### CODE REVIEW

File: mermaid-mode.el [hunk 1]
```diff
@@ -57,16 +59,31 @@
...
-      '(("graph \\|subgraph \\|end\\|sequenceDiagram\\|loop \\|alt \\|else \\|opt" . font-lock-keyword-face)
-        ("---\\|-?->*\\+?" . font-lock-function-name-face)
-        ("LR\\|TD\\|participant \\|Note" . font-lock-constant-face)))
+      '(("graph \\|subgraph \\|end \\|pie \\|gantt \\|classDiagram \\|stateDiagram \\|title \\|sequenceDiagram\\|loop \\|alt \\|else \\|opt" . font-lock-keyword-face)
+        ("---\\|-?->*\\+?\\|==>\\|===" . font-lock-function-name-face)
+        ("LR\\|TD\\|TB\\|RL\\|DT\\|BT\\|participant \\|Note" . font-lock-constant-face)))
+
+(defvar org-babel-default-header-args:mermaid
+  '((:results . "file") (:exports . "results"))
+  "Default arguments for evaluatiing a mermaid source block.")
```
Comment (abrochard) [inline mermaid-mode.el]: `evaluatiing` typo

File: mermaid-mode.el [hunk 2]
```diff
@@ -57,16 +59,31 @@
...
+
+(defun org-babel-execute:mermaid (body params)
```
Comment (abrochard) [inline mermaid-mode.el]: how does this work? it is going to be automatically called on org src block with mermaid mode?

**Comment by abrochard** [comment]
Thank you for your PR! I left you a quick nitpick and a general question. Will merge soon

File: mermaid-mode.el (hunk 2 above)
Comment (nanjj) [reply mermaid-mode.el]: when you call org-publish-current-file, it will do tranform and replace the block with the image it generated

**Pull request merged**